pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -189,18 +204,34 @@ func mergeRoutesWeightedClusters(originalRoutes, latestRoutes []*RouteWeightedCl
 	return originalRoutes
 }
 
-// subset returns true if the second array is completely contained in the first array
-func subset(first, second []string) bool {
-	set := make(map[string]bool)
-	for _, value := range first {
-		set[value] = true
+// subset union returns true if either of the slices is a subset of the other and returns a unified slice
+func subsetUnion(first, second []string) (bool, []string) {
```

Comment:
If the function returns a string slice comprising of the union of the 2 input slices only when the slices are subsets of each other, then the bool return is unnecessary because an empty slice returned would imply there is no union?

Reply:
makes sense, I removed the bool return